core/modules/replica/tools/qserv-replica-master.cc

```diff
@@ -35,715 +40,322 @@
+     *
+     * @param logger
+     *   the name of a logger
+     */
+    static Ptr create(int argc,
+                      const char* const argv[],
+                      std::string const& description,
+                      std::string const& logger) {
+        return Ptr(
+            new MasterControllerApp(
+                argc,
+                argv,
+                description,
+                logger
+            )
+        );
+    }
 
-    /// no parent for any job initited by the application
-    std::string const parentJobId;
+    // Default construction and copy semantics are prohibited
 
-    /// Force Qserv synchronization if 'true'
-    bool forceQservSync;
+    MasterControllerApp() = delete;
+    MasterControllerApp(MasterControllerApp const&) = delete;
+    MasterControllerApp& operator=(MasterControllerApp const&) = delete;
 
-    /// Permanently delete workers if set to 'true'
-    bool permanentDelete;
+    ~MasterControllerApp() override = default;
 
-    // Context strings for threads
+protected:
 
-    std::string healthMonitorContext;
-    std::string replicationLoopContext;
+    /**
+     * @see MasterControllerApp::create()
+     */
+    MasterControllerApp(int argc,
+                        const char* const argv[],
+                        std::string const& description,
+                        std::string const& logger)
+        :   Application(
+                argc,
+                argv,
+                description,
+                true /* injectDatabaseOptions */,
+                true /* boostProtobufVersionCheck */,
+                true /* enableServiceProvider */
+            ),
+            _healthProbeIntervalSec  (60),
+            _replicationIntervalSec  (60),
+            _workerResponseTimeoutSec(60),
+            _workerEvictTimeoutSec   (3600),
+            _qservSyncTimeoutSec     (60),
+            _numReplicas             (0),
+            _numIter                 (0),
+            _purge          (false),
+            _forceQservSync (false),
+            _permanentDelete(false),
+            _failed         (false),
+            _log(LOG_GET(logger)) {
+
+        // Configure the command line parser
+
+        parser().option(
+            "health-probe-interval",
+            "interval (seconds) between iterations of the health monitoring probes",
+            _healthProbeIntervalSec
+        ).option(
+            "replication-interval",
+            "interval (seconds) between running the linear sequence of"
+            " actions: check - fix-up - replicate - re-balance",
+            _replicationIntervalSec
+        ).option(
+            "worker-response-timeout",
+            "maximum number of seconds to wait before giving up"
+            " on worker probes when checking for workers",
+            _workerResponseTimeoutSec
+        ).option(
+            "worker-evict-timeout",
+            "the maximum number of seconds to allow troubles workers to recover"
```

Comment:
'troubles' is not the right word here. 'failed', 'slow', 'workers having trouble'?

Reply:
👍